lib/Connector/CapConnector.php

```
@@ -294,6 +287,62 @@ private function processCapData(DataProviderInterface $dataProvider): void
+     *
+     * @param DataProviderInterface $dataProvider
+     * @param Carbon $cacheExpiresAt
+     *
+     * @return string
+     * @throws GeneralException|GuzzleException
+     */
+    private function fetchCapAlertFromUrl(DataProviderInterface $dataProvider, Carbon $cacheExpiresAt): string
+    {
+        $emergencyAlertUrl = $dataProvider->getProperty('emergencyAlertUri');
+
+        $cache = $this->pool->getItem('/emergency-alert/cap/' . md5($emergencyAlertUrl));
+        $data = $cache->get();
+
+        if ($cache->isMiss()) {
+            $cache->lock();
+            $this->getLogger()->debug('Getting CAP data from CAP Feed');
+
+            $httpOptions = [
+                'timeout' => 20, // Wait no more than 20 seconds
+            ];
+
+            try {
+                // Make a GET request to the CAP URL using Guzzle HTTP client with defined options
+                $response = $dataProvider
+                    ->getGuzzleClient($httpOptions)
+                    ->get($emergencyAlertUrl);
+
+                $this->getLogger()->debug('CAP Feed: uri: ' . $emergencyAlertUrl . ' httpOptions: '
+                        . json_encode($httpOptions));
+
+                // Get the response body as a string
+                $data = $response->getBody()->getContents();
+
+                // Cache
+                $cache->set($data);
+                $cache->expiresAt($cacheExpiresAt);
+                $this->pool->saveDeferred($cache);
+            } catch (RequestException $e) {
+                // Log the error with a message specific to CAP data fetching
+                $this->getLogger()->error('Unable to reach the CAP feed URL: '
+                    . $emergencyAlertUrl . ' Error: ' . $e->getMessage());
+
+                // Throw a more specific exception message
+                throw new GeneralException('Failed to retrieve CAP data from the specified URL.');
```

Should we surface this error into the layout editor using `->addError()` ?